### DISCUSSION TITLE
API + MODULE only, how to connect to Database ?

### DISCUSSION THREAD

**Discussion by Yamakasi** [osm-search/Nominatim#2199] [Q&A]
As I'm stripping off my nominatim install I have build/installed (yes master) the API and MODULE so I should be able to connect to PostgreSQL only. I will import using osm2pgsql which, I think, suits me the most.

Everything is installed in `/usr/local/lib64/nominatim` but I cannot find any way to put my database server settings in. I see a function `generateDSN()` in `/usr/local/lib64/nominatim/lib-php/DB.php` but that's about it.

I cannot find anything about it in the docs, neither examples online or github.

Any help/ideas here ?

**Comment by lonvia** [comment]
Configuration is described in the [installation docs for master](https://nominatim.org/release-docs/develop/admin/Import/#configuration-setup-in-env).

I don't know what exactly you mean with 'I will import using osm2pgsql which, I think, suits me the most.'. Nominatim is an integrated software that has an import process to build up the data structures it needs. You can't simply swap that out with some other tool. You shouldn't even switch out the bundled osm2pgsql with another version of your own choosing. There is no guarantee it will work.

On a more general note: you seem to be on a mission to strip down software in quite bizarre ways. Before you ask further questions, could you please describe what you plan to do. Because at the moment it looks like you are way out in "this is not supported in any way, you are on your own" country.

**Reply by Yamakasi** [reply]
> Configuration is described in the [installation docs for master](https://nominatim.org/release-docs/develop/admin/Import/#configuration-setup-in-env).
> 

Yes but this doesn't make much clear. "The Nominatim server" is not one thing, yes if you put everything on one box which is very clumsy; especially if you want flexible failover for reading your data and easy upgrades. The question here is, why can I build the API here seperate but there is no file anymore for the settings ?

I have checked the ENV variables but nothing does apply to the API (PHP) it seems, or does it use `nominatim/db/connection.py` ?

> I don't know what exactly you mean with 'I will import using osm2pgsql which, I think, suits me the most.'. Nominatim is an integrated software that has an import process to build up the data structures it needs. You can't simply swap that out with some other tool. You shouldn't even switch out the bundled osm2pgsql with another version of your own choosing. There is no guarantee it will work.
> 

No worries, version control will be monitored. osm2pgsql works great by itself, it has a great way of updating the diffs on a daily/weekly base and it can be runned seperatly on a seperate box. Why put all heavy stuff on one overrated server as a SPOF ?

> On a more general note: you seem to be on a mission to strip down software in quite bizarre ways. Before you ask further questions, could you please describe what you plan to do. Because at the moment it looks like you are way out in "this is not supported in any way, you are on your own" country.

It's not bizarre, I develop my own and maintain for HA clusters with minimal services per container so keep it flexible. It doesn't need to be supported but why does the build process has options to build specific parts anyways ? 

**I'm just trying to figure out the components by itself to see how we can scale it better.**

**Comment by Yamakasi** [comment]
Any update on this ? I'm not able to find real detailed information about parts of the build when not all together.
